### PR TITLE
✨ Finish review survey sumary #496

### LINKED ISSUE #496: Show summary with correct and incorrect answers when survey is complete
When user

- Select a survey
- answers all questions
- user should see how many answers were correct 
- **user should see correct answers**
- **user should see incorrect answers**

### PR DESCRIPTION
**Pull request by iurynogueira**
fix #496

# With this we have MVP

- [ ] **Model testing.**
- [ ] **Request testing and swagger doc update.**
- [x] **System testing.**
- [ ] **No tests required.**

### CODE REVIEW

File: frontend/src/api/services/answer_survey.test.ts [hunk 1]
```diff
@@ -77,4 +80,52 @@ describe('AnswerSurveyService', () => {
       });
     });
   });
+
+  describe('get method', () => {
```
Comment (edimossilva) [inline frontend/src/api/services/answer_survey.test.ts:84]: Usually just
```suggestion
  describe('get', () => {
```

File: frontend/src/api/services/answer_survey.test.ts [hunk 2]
```diff
@@ -77,4 +80,52 @@ describe('AnswerSurveyService', () => {
...
+        instance = new AnswerSurveyService(httpClient as any);
+
+        res = await instance.get(mockAnswerSurveyGet);
+      });
+
+      it('should call get method once', () => {
+        expect(httpClient.get).toHaveBeenCalled();
+      });
+
+      it('should return expected data', () => {
```
Comment (edimossilva) [inline frontend/src/api/services/answer_survey.test.ts:103]: is expected data `expect(res).toEqual(undefined);`?

File: frontend/src/components/sumary-result/sumary-result.test.tsx [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+    expect(screen.getByText('Finish!')).toBeInTheDocument();
+  });
+
+  it('should show the result of survey', () => {
+    render(<SumaryResult {...answerSurvey} />);
+    expect(screen.getByText('1 correct')).toBeInTheDocument();
+    expect(screen.getByText('1 incorrect')).toBeInTheDocument();
+  });
+
+  it('should show questions corrects and incorrects', () => {
```
Comment (edimossilva) [inline frontend/src/components/sumary-result/sumary-result.test.tsx:78]:
```suggestion
  it('should show correct and incorrect questions', () => {
```

**Review by edimossilva — COMMENTED** [review]
Please check the spec that I commented.

Congratz for doing it so fast

File: frontend/src/api/services/answer_survey.test.ts (hunk 2 above)
Comment (iurynogueira) [reply frontend/src/api/services/answer_survey.test.ts:103]: Yep!

` public async get(
    answer_survey: AnswerSurveyGet
  ): Promise<AxiosResponse<AnswerSurveyReceive | undefined>> {
    try {
      const response = await this.httpClient.get<AnswerSurveyReceive>(
        `api/v1/answers_surveys/${answer_survey.id}`
      );

      return response;
    } catch (error) {
      errorHandler(error);
    }

    return undefined;
  }`

The service can return undefined if try not called.

**Comment by iurynogueira** [comment]
> 

Fixed! Thanks! <3

**Review by edimossilva — APPROVED** [review]

**Pull request merged**